python/pyarrow/lib.pyx

```diff
@@ -120,6 +120,8 @@ UnionMode_DENSE = _UnionMode_DENSE
 
 def _pc():
     import pyarrow.compute as pc
+    from pyarrow import _exec_plan
+    pc._exec_plan = _exec_plan
```

Comment:
In theory, pyarrow.dataset is not necessarily enabled / built, which would mean that then all pyarrow.compute based methods would also fail? 
(although I am not sure whether we still have (nightly) builds that exercise this option..)

Reply:
Good point, I'll try to put in place machinery to make sure `_exec_plan` can be imported even when datasets are disabled.

Reply:
addressed